Scope Tags to Content Types (#1505)
nautobot/nautobot: nautobot/docs/release-notes/version-1.3.md

Nautobot's REST API now supports multiple versions, which may requested by modifying the HTTP Accept header on any requests sent by a REST API client. Details are in the [REST API documentation](../rest-api/overview.md#versioning), but in brief:

> - The REST API endpoints that are versioned in the 1.3.0 release are
>   - `/api/extras/jobs/` listing endpoint
>   - `/api/extras/tags/` create/put/patch endpoint
>   - `/api/ipam/ip-addresses/` endpoints
> - All others endpoints are currently un-versioned. However, over time more versioned REST APIs will be developed, so this is important to understand for all REST API consumers.

Reviewer: For some reason, ReadTheDocs has a different expectation on indenting of nested lists than local rendering with `mkdocs` alone - the below is needed for correct rendering on RTD. See also #1557.

```suggestion
- The REST API endpoints that are versioned in the 1.3.0 release are
    - `/api/extras/jobs/` listing endpoint
    - `/api/extras/tags/` create/put/patch endpoint
    - `/api/ipam/ip-addresses/` endpoints
- All others endpoints are currently un-versioned. However, over time more versioned REST APIs will be developed, so this is important to understand for all REST API consumers.
```

- If a REST API client does not request a specific REST API version (in other words, requests `Accept: application/json` rather than `Accept: application/json; version=1.3`) the API behavior will be compatible with Nautobot 1.2, at a minimum for the remainder of the Nautobot 1.x release cycle.
- The API behavior may change to a newer default version in a Nautobot major release (e.g. 2.0).

> - To request an updated (non-backwards-compatible) API endpoint, an API version must be requested corresponding at a minimum to the Nautobot `major.minor` version where the updated API endpoint was introduced (so to interact with the new Jobs REST API, `Accept: application/json; version=1.3`).

Reviewer: Probably would be good to add a note about the Tag changes as a separate heading under `Release Overview` > `Changed`. Specifically it would be good to call out that a tag created programmatically via the ORM without assigning any content-types will not be applicable to any model until content-types are assigned to it.

Nautobot's REST API now supports multiple versions, which may requested by modifying the HTTP Accept header on any requests sent by a REST API client. Details are in the [REST API documentation](../rest-api/overview.md#versioning), but in brief:

> - The REST API endpoint that are versioned in the 1.3.0 release are the `/api/extras/jobs/` listing endpoint and `/api/extras/tags/` create/put/patch endpoint, as described above. All others are currently un-versioned. However, over time more versioned REST APIs will be developed, so this is important to understand for all REST API consumers.

Reviewer 2: ```suggestion
- The REST API endpoints that are versioned in the 1.3.0 release are
  - `/api/extras/jobs/` listing endpoint
  - `/api/extras/tags/` create/put/patch endpoint
  - `/api/ipam/ip-addresses/` endpoints
- All others endpoints are currently un-versioned. However, over time more versioned REST APIs will be developed, so this is important to understand for all REST API consumers.
```

Similar to the existing `extras.run_job` permission, a new `extras.approve_job` permission is now enforced by the UI and the REST API when approving scheduled jobs. Only users with this permission can approve or deny approval requests; additionally such users also now require the `extras.view_scheduledjob`, `extras.change_scheduledjob`, and `extras.delete_scheduledjob` permissions as well.

> #### Tags without content_types ([#1505](https://github.com/nautobot/nautobot/pull/1505))
> 
> Note that Tags created programmatically via the ORM without assigning a content_types will not be applicable to any model until content-types are assigned to it.
> 
> #### OpenAPI 3.0 REST API documentation ([#595](https://github.com/nautobot/nautobot/issues/595))
> 
> The online REST API Swagger documentation (`/api/docs/`) has been updated from OpenAPI 2.0 format to OpenAPI 3.0 format and now supports Nautobot's [REST API versioning](#rest-api-versioning-1465) as described above. Try `/api/docs/?api_version=1.3` as an example.

Reviewer: ```suggestion
#### OpenAPI 3.0 REST API documentation ([#595](https://github.com/nautobot/nautobot/issues/595))

The online REST API Swagger documentation (`/api/docs/`) has been updated from OpenAPI 2.0 format to OpenAPI 3.0 format and now supports Nautobot's [REST API versioning](#rest-api-versioning-1465) as described above. Try `/api/docs/?api_version=1.3` as an example.

#### Tag restriction by content-type ([#872](https://github.com/nautobot/nautobot/issues/872))

When created, a `Tag` can be associated to one or more model content-types using a many-to-many relationship. The tag will then apply only to models belonging to those associated content-types.

For users migrating from an earlier Nautobot release, any existing tags will default to being enabled for all content-types for compatibility purposes. Individual tags may subsequently edited to remove any content-types that they do not need to apply to.

Note that a Tag created programmatically via the ORM without assigning any `content_types` will not be applicable to any model until content-types are assigned to it.
```
